### ISSUE TITLE
'cdcPipelineConfigurator': IllegalArgumentException: readerName must not be null

### ISSUE THREAD

**Issue by RaccoonDev** [eventuate-foundation/eventuate-cdc#18]
Hi,

Using TODO sample application I faced a problem that CDC does not do its job. All messages are not published to kafka. So, I updated the docker images to the latest versions (0.21.3) and changed version for eventuate tram to 0.11.1.RELEASE.

With that set up I'm getting CDC service failing to start with error: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cdcPipelineConfigurator': Invocation of init method failed; nested exception is java.lang.IllegalArgumentException: readerName must not be null

Is the issue caused by misconfiguration? Or I messed up with versions?
Please help.

**Comment by cer** [comment]
A few things.

0. Which Todo application; Eventuate Tram (https://github.com/eventuate-tram/eventuate-tram-examples-java-spring-todo-list) or Eventuate Local (https://github.com/eventuate-examples/eventuate-examples-java-spring-todo-list)

1. It should not have been necessary to have upgraded the CDC.  I suspect there is something other problem.

2. For configuring the newer CDC, see this example https://github.com/microservices-patterns/ftgo-application/blob/master/docker-compose.yml#L25 and look at the manual: http://eventuate.io/docs/manual/eventuate-tram/latest/cdc-configuration.html#cdc-configuration

**Comment by RaccoonDev** [comment]
Thanks for pointing me to the docs!

TL;DR: Configuration below allowed me to successfully run CDC service outsize of the docker-compose setup.

You're right. That was my original problem with configuration. DOCKER_HOST_IP is kind of not applicable for my Mac OS setup. So, the value from docker was returned as an empty string. So, that led to misconfigured Kafka which ADVERTISED_HOST_NAME. Actually writing that, I realized that that might not be the real cause of why CDC was not running in the first place from the container.

To figure out what exactly is happening I ran CDC service from outside of the docker-compose system. Successfully configured it to connect to the infrastructure created by docker-compose. 

I'm excited to dig deeper into eventuate framework, services, and approaches. So, thank for the help again.

To answer my original question. The configuration that was necessary for my CDC service setup:

```
management.endpoint.health.show-details=ALWAYS
management.endpoints.web.exposure.include=prometheus,health
logging.level.io.eventuate.tram=INFO
spring.datasource.url=jdbc:mysql://Dmytros-MacBook-Pro-2.local/eventuate
spring.datasource.username=mysqluser
spring.datasource.password=mysqlpw
spring.datasource.driver.class.name=com.mysql.jdbc.Driver
eventuatelocal.kafka.bootstrap.servers=Dmytros-MacBook-Pro-2.local:9092
eventuatelocal.zookeeper.connection.string=Dmytros-MacBook-Pro-2.local:2181
eventuatelocal.cdc.db.user.name=root
eventuatelocal.cdc.db.password=rootpassword
eventuatelocal.cdc.mysql.binlog.client.unique.id=1234567890
eventuatelocal.cdc.source.table.name=message
eventuatelocal.cdc.reader.name=customcdcreader
eventuatelocal.cdc.read.old.debezium.db.offset.storage.topic=true
```

Thank you,
D
